docs: first pass at working with networks page (#830)
OpenFreeEnergy/openfe: docs/guide/results/working_with_networks.rst

Working with networks of results
================================

After running a **network** of free energy calculations,

> we often want to consider analysing the corresponding network of results.

Reviewer: ```suggestion
we often want to analyse the corresponding network of results.
```
I think "consider" may not be necessary?

.. _userguide_MLE:

Converting relative results to absolute estimates
-------------------------------------------------

> When a network of relative free energies have been calculated,

Reviewer: ```suggestion
When a network of relative free energies has been calculated,
```

> a commonly performed task is to transform these pairwise estimations of **relative** free energies

Reviewer: Technically, we never obtain actual "free energies" but always "free energy differences" (so we don't get G but always Delta G or Delta Delta G). I know that always sounds very wordy so I'm not sure if it needs to be changed here or if we can leave it as is...

> a commonly performed task is to transform these pairwise estimations of **relative** free energies differences (:math:`\Delta \Delta G`)

Reviewer: ```suggestion
a commonly performed task is to transform these pairwise estimations of **relative** free energy differences (:math:`\Delta \Delta G`)
```

> into a **absolute** free energies.

Reviewer: Either
```suggestion
into **absolute** free energies.
```
or
```suggestion
into a **absolute** free energy.
```

This is done using a maximum likelihood estimator (MLE) [1]_,
as implemented in the `cinnabar`_ package.
This approach uses the matrix of relative pairwise measurements and their uncertainties,

> to estimate the overall ranking of

Reviewer: ranking of ...? (ligands? results?)

To use this approach the network of pairwise measurements needs to be fully connected,
i.e. there should be a way to trace a path along pairwise measurements between any two nodes (ligands) on the network.

.. note::

>    The results of a MLE estimation will have a **mean** of 0.0,

Reviewer: Do we have a code snippet somewhere (maybe in a cookbook?) that shows how one can shift the data?

Author: not really, I need to add in this shifting to the cinnabar docs (and our gather command probably...)

   meaning that there will be some estimates with positive values and some estimates with negative values.
   These predictions (:math:`\Delta G_{pred}`) can be shifted to match the magnitude of the experimental data,
   to satisfy the below equation where the sum is performed over N molecules that have experimental data (:math:`\Delta G_{exp}`) [2]_.

   .. math::

      \sum_i^N \Delta G^i_{exp} = \sum_i^N \Delta G^i_{pred}

Gathering using the command line
--------------------------------

After running calculations using the :ref:`quickrun command <userguide_quickrun>`,
the :ref:`openfe gather <cli_gather>` command offers a way to collate information across many different individual
simulations and prepare a table of results.

> The tool offers a summary of the relative binding affinities,

Reviewer: Would it make sense to add the command line option here?, e.g.
```suggestion
The tool offers a summary of the relative binding affinities (`--report ddg`),
```
and then in the next line ``--report dg``

> The tool offers a summary of the relative binding affinities (`--report ddg`),

Reviewer: ```suggestion
The tool offers a summary of the relative binding affinities (``--report ddg``),
```

> or their :ref:`corresponding MLE values <userguide_MLE>`.

Reviewer: ```suggestion
or their :ref:`corresponding MLE values <userguide_MLE>` (`--report dg`).
```

Using cinnabar directly
-----------------------

The `cinnabar`_ package can be used from within Python to manipulate networks of free energy estimates.
A tutorial on using this is provided here :ref:`here <tutorials/plotting_with_cinnabar.nblink>`

See also
--------

> * For handling the results of a single calculation, consult :ref:`userguide_individual_results`

Reviewer: ```suggestion
For handling the results of a single calculation, please consult :ref:`userguide_individual_results`
```
Looks a little bit weird to have a single bullet point here...
